rx_signal_processing/dsp.cu

```diff
@@ -109,20 +109,45 @@ namespace {
     }
   }
 
+  /**
+   * @brief      Beamforms the final samples
+   *
+   * @param      filtered_samples         A flat vector containing all the filtered samples.
+   * @param      beamformed_samples_main  A vector where the beamformed main samples are placed.     
```

Comment:
can we be more clear in this docstring that the samples from each array are also combined here? maybe 'beamformed and combined main array samples' and similarly for intf

Reply:
What do you mean by combined? do you mean they are combined into a flat vector, or that all antenna samples are combined into one beam?

Reply:
I mean that it could be possible to do the beam phasing on the individual antenna data without combining into a single 'sample set' per beam. The docstring does say you're using matrix multiplication so it could be implied that this is happening but I think you could be more clear in describing this output.